scipy/stats/_continuous_distns.py

```diff
@@ -3247,6 +3247,142 @@ def _ppf(self, q, k, s):
 mielke = mielke_gen(a=0.0, name='mielke')
 
 
+class kappa4_gen(rv_continuous):
+    """Kappa 4 parameter distribution
+
+    %(before_notes)s
+
+    Notes
+    -----
+    The probability density function for kappa4 is::
+
+        kappa4.pdf(x, h, k) = (1.0 - k*x)**(1.0/k - 1.0)*
+                              (1.0 - h*(1.0 - k*x)**(1.0/k))**(1.0/h-1)
+
+    if ``h`` and ``k`` are not equal to 0.
+
+    kappa4 takes ``h`` and ``k`` as shape parameters, skew and
+    kurtosis respectively.
+
+    The kappa4 distribution return other distributions when certain
+    ``h`` and ``k`` values are used.
+
+    +------+-------------+--------------+---------------+
+    | h    | k=0.0       | k=1.0        | -2<=k<=2      |
+    +======+=============+==============+===============+
+    |  1.0 | Exponential | Uniform      | Generalized   |
+    |      |             |              | Pareto        |
+    +------+-------------+--------------+---------------+
+    |  0.0 | Gumbel      | Reverse      | Generalized   |
+    |      |             | Exponential  | Extreme Value |
+    |      |             | (note:not in |               |
+    |      |             | scipy)       |               |
+    +------+-------------+--------------+---------------+
+    | -1.0 | Logistic    |              | Generalized   |
+    |      |             |              | Logistic      |
+    +------+-------------+--------------+---------------+
+
+    %(after_notes)s
+
+    %(example)s
+
+    """
+    def _argcheck(self, h, k):
+        hc = np.atleast_1d(h)[0]
+        self.hc = hc
+        kc = np.atleast_1d(k)[0]
+        self.kc = kc
```

Comment:
I've to admit I don't like this. Granted, there does not seem to be a way of correctly handling e.g. `h=[0, 1]` --- the generic code path does not handle well the case where `a` or `b` are shape-dependent. 

Nonetheless, attaching attributes to the instance like this feels like a hack. More importantly, if and when there is work to allow handling array-valued support bounds, this hack will need to be undone *in addition to* fixing the generic code paths.

At the very least, these extra attributes must have a leading underscore so that they are not public.

All in all, my gut feeling is that it's better to use a helper along the lines of https://github.com/scipy/scipy/commit/0ffb413cf8184ba20d7c33cd7e1b1027d4c03473
(this `_lazyselect` is not very well tested, so can be buggy, too).